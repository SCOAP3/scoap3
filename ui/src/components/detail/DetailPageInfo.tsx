import React from "react";
import moment from "moment";
import { Divider } from "antd";

import { JournalInfo, ArticleIdentifier, License, Result } from "@/types";
import { resolveIdentifierLink } from "@/utils/utils";
import FulltextFiles from "../shared/FulltextFiles";
import PublicationInfo from "../shared/PublicationInfo";

interface DetailPageInfoProps {
  article: Result;
}

const DetailPageInfo: React.FC<DetailPageInfoProps> = ({ article }) => {
  const { artid, publisher } =
    (article?.publication_info?.[0] as JournalInfo) || {};

  const renderIdentifierLinks = (identifiers: ArticleIdentifier[]) => {
    return identifiers?.map((identifier) => (
      <div key={identifier?.identifier_type}>
        <dt>{identifier?.identifier_type}:</dt>
        {identifier?.identifier_type === "arXiv" && (
          <dd>
            {article?.article_arxiv_category?.map(
              (category) => category?.primary && category?.category
            )}
          </dd>
        )}
        <dd>
          <a href={resolveIdentifierLink(identifier)}>
            {identifier?.identifier_value}
          </a>
        </dd>
      </div>
    ));
  };

  const renderLicenses = (licenses: License[]) => {
    return licenses?.map((licence) => (
      <a href={licence?.url} key={licence?.name}>
        {licence?.name}
      </a>
    ));
  };

  return (
    <dl className="m-0 pb-5">
      <dt>Published on:</dt>
      <dd>{moment(article?.publication_date).format("DD MMMM YYYY")}</dd>
      <dt>Created on:</dt>
      <dd>{moment(article?._created_at).format("DD MMMM YYYY")}</dd>
      <dt>Publisher:</dt>
      <dd>{publisher}</dd>
      <dt>Published in:</dt>
      <dd>
        <PublicationInfo data={article?.publication_info?.[0]} page="detail" />
      </dd>
      {artid && <dd>Article ID: {artid}</dd>}
      {renderIdentifierLinks(article?.article_identifiers)}
      <dt>Copyrights:</dt>
      <dd>{article?.copyright?.[0]?.statement}</dd>
      <dt>Licence:</dt>
      <dd>{renderLicenses(article?.related_licenses)}</dd>

      <Divider />

      <dt>Fulltext files:</dt>
      <dd className="flex mt-2">
        <FulltextFiles files={article?.related_files} />
      </dd>
    </dl>
  );
};

export default DetailPageInfo;
