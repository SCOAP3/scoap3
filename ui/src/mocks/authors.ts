export const authors = [
  {
    first_name: "John",
    last_name: "Doe",
    affiliations: [
      { value: "Affiliation 1" },
      { value: "Affiliation 2" },
    ],
    orcid: "1234-5678-9101-1126",
  },
  {
    first_name: "Jane",
    last_name: "Smith",
    affiliations: [
      { value: "Affiliation 3" },
    ],
    orcid: "9876-5432-1098-7658",
  },
  {
    first_name: "John",
    last_name: "Smith",
    orcid: "1234-5678-9101-1120",
  },
  {
    first_name: "Jane",
    last_name: "Doe",
    orcid: "9876-5432-1098-7651",
  },
  {
    first_name: "Mary",
    last_name: "Doe",
    orcid: "1234-5678-9101-1121",
  },
  {
    first_name: "Michael",
    last_name: "Smith",
    orcid: "9876-5432-1098-7653",
  },
  {
    first_name: "Tom",
    last_name: "Jones",
    orcid: "9876-5432-1098-7652",
  }
];
