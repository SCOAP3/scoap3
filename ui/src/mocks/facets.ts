export const facets = {
  _filter_publication_year: {
    doc_count: 123,
    publication_year: {
      buckets: [
        {
          key: 1388534400000,
          doc_count: 2137,
        },
      ],
    },
  },
  _filter_country: {
    doc_count: 123,
    country: {
      buckets: [
        {
          key: "Poland",
          doc_count: 2137,
        },
      ],
    },
  },
  _filter_journal: {
    doc_count: 123,
    journal: {
      buckets: [
        {
          key: "Journal of High Energy Physics",
          doc_count: 666,
        },
      ],
    },
  },
};
