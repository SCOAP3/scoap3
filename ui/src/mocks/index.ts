export { facets } from  "./facets";
export { results } from  "./results";
export { query } from "./query";
export { record } from "./record";
export { authors } from "./authors";
