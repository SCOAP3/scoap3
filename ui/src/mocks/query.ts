export const query = {
  page: 1,
  page_size: 20,
  journal: "Journal of High Energy Physics",
  publication_year__range: "2014__2023",
  country: "Sweden",
};
