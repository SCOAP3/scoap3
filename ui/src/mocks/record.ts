export const record = {
  related_files: [
    {
      file: "https://scoap3-dev-backend.s3.cern.ch/media/files/10913/10.1007/JHEP06(2015)171.xml",
      created: "2023-10-11T17:10:45.027275+00:00",
      updated: "2023-10-11T17:10:45.027283+00:00",
    },
    {
      file: "https://scoap3-dev-backend.s3.cern.ch/media/files/10913/10.1007/JHEP06(2015)171_a.pdf",
      created: "2023-10-11T17:10:45.030311+00:00",
      updated: "2023-10-11T17:10:45.030318+00:00",
    },
  ],
  _created_at: "2018-04-30T15:04:40.805229+00:00",
  copyright: [
    {
      statement: "The Author(s)",
      year: null,
      holder: "",
    },
  ],
  article_identifiers: [
    {
      identifier_type: "DOI",
      identifier_value: "10.1007/JHEP06(2015)171",
    },
    {
      identifier_type: "arXiv",
      identifier_value: "1504.07579",
    },
  ],
  related_materials: [],
  title:
    "Strings from 3D gravity: asymptotic dynamics of AdS 3 gravity with free boundary conditions",
  related_licenses: [
    {
      url: "http://creativecommons.org/licenses/by/4.0/",
      name: "CC-BY-4.0",
    },
  ],
  id: 10913,
  subtitle: "",
  publication_date: "2015-06-25",
  publication_info: [
    {
      journal_volume: "",
      journal_issue: "",
      journal_title: "Journal of High Energy Physics",
      page_start: "171",
      page_end: "189",
      artid: "123456",
      volume_year: "2015",
      journal_issue_date: null,
      publisher: "Springer/SISSA",
    },
  ],
  article_arxiv_category: [
    {
      category: "hep-th",
      primary: true,
    },
    {
      category: "gr-qc",
      primary: false,
    },
  ],
  acceptance_date: null,
  abstract:
    "Pure three-dimensional gravity in anti-de Sitter space can be formulated as an SL(2 , R ) Ã— SL(2 , R ) Chern-Simons theory, and the latter can be reduced to a WZW theory at the boundary. In this paper we show that AdS 3 gravity with free boundary conditions is described by a string at the boundary whose target spacetime is also AdS 3 . While boundary conditions in the standard construction of Coussaert, Henneaux, and van Driel are enforced through constraints on the WZW currents, we find that free boundary conditions are partially enforced through the string Virasoro constraints.",
  first_online_date: null,
  reception_date: null,
  authors: [
    {
      first_name: "Luis",
      last_name: "Apolo",
      affiliations: [
        {
          value:
            "Department of Physics & The Oskar Klein Centre, Stockholm University, AlbaNova University Centre, SE-106 91, Stockholm, Sweden",
          organization: "",
          country: {
            code: "SE",
            name: "Sweden",
          },
        },
      ],
    },
    {
      first_name: "Bo",
      last_name: "Sundborg",
      affiliations: [
        {
          value:
            "Department of Physics & The Oskar Klein Centre, Stockholm University, AlbaNova University Centre, SE-106 91, Stockholm, Sweden",
          organization: "",
          country: {
            code: "SE",
            name: "Sweden",
          },
        },
      ],
    },
  ],
};
