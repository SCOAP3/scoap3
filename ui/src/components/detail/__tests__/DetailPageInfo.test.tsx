import React from "react";
import { render, screen } from "@testing-library/react";

import DetailPageInfo from "../DetailPageInfo";
import { record } from '@/mocks/record';

describe("DetailPageInfo", () => {
  it("renders detail page info correctly", () => {
    const { container } = render(<DetailPageInfo article={record} />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText("Published on:")).toBeInTheDocument();
    expect(screen.getByText("25 June 2015")).toBeInTheDocument();
    expect(screen.getByText("Created on:")).toBeInTheDocument();
    expect(screen.getByText("30 April 2018")).toBeInTheDocument();
    expect(screen.getByText("Springer/SISSA")).toBeInTheDocument();
    expect(screen.getByText("Published in:")).toBeInTheDocument();
    expect(screen.getByText("Journal of High Energy Physics")).toBeInTheDocument();
    expect(screen.getByText("(2015)")).toBeInTheDocument();
    expect(screen.getByText("Pages 171-189")).toBeInTheDocument();
    expect(screen.getByText("DOI:")).toBeInTheDocument();
    expect(screen.getByText("10.1007/JHEP06(2015)171")).toBeInTheDocument();
    expect(screen.getByText("arXiv:")).toBeInTheDocument();
    expect(screen.getByText("hep-th")).toBeInTheDocument();
    expect(screen.getByText("1504.07579")).toBeInTheDocument();
    expect(screen.getByText("Copyrights:")).toBeInTheDocument();
    expect(screen.getByText("The Author(s)")).toBeInTheDocument();
    expect(screen.getByText("Licence:")).toBeInTheDocument();
    expect(screen.getByText("CC-BY-4.0")).toBeInTheDocument();
    expect(screen.getByText("Fulltext files:")).toBeInTheDocument();
    expect(screen.getByText("XML")).toBeInTheDocument();
    expect(screen.getByText("PDF/A")).toBeInTheDocument();
  });
});
