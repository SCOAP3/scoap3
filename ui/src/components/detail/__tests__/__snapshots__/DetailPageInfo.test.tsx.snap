// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPageInfo renders detail page info correctly 1`] = `
<div>
  <dl
    class="m-0 pb-5"
  >
    <dt>
      Published on:
    </dt>
    <dd>
      25 June 2015
    </dd>
    <dt>
      Created on:
    </dt>
    <dd>
      30 April 2018
    </dd>
    <dt>
      Publisher:
    </dt>
    <dd>
      Springer/SISSA
    </dd>
    <dt>
      Published in:
    </dt>
    <dd>
      <p
        class="publication-info-detail false inline-block mb-1"
      >
        <span
          class="publication-info-title"
        >
          Journal of High Energy Physics
        </span>
        <span
          class="publication-info-volume_year"
        >
           (2015)
        </span>
        <span
          class="publication-info-pages"
        >
          Pages 171-189
        </span>
      </p>
    </dd>
    <dd>
      Article ID:
      123456
    </dd>
    <div>
      <dt>
        DOI
        :
      </dt>
      <dd>
        <a
          href="https://doi.org/10.1007/JHEP06(2015)171"
        >
          10.1007/JHEP06(2015)171
        </a>
      </dd>
    </div>
    <div>
      <dt>
        arXiv
        :
      </dt>
      <dd>
        hep-th
      </dd>
      <dd>
        <a
          href="https://arxiv.org/abs/1504.07579"
        >
          1504.07579
        </a>
      </dd>
    </div>
    <dt>
      Copyrights:
    </dt>
    <dd>
      The Author(s)
    </dd>
    <dt>
      Licence:
    </dt>
    <dd>
      <a
        href="http://creativecommons.org/licenses/by/4.0/"
      >
        CC-BY-4.0
      </a>
    </dd>
    <div
      class="ant-divider css-dev-only-do-not-override-2q8sxy ant-divider-horizontal"
      role="separator"
    />
    <dt>
      Fulltext files:
    </dt>
    <dd
      class="flex mt-2"
    >
      <a
        class="file-big mr-2"
        download=""
        href="https://scoap3-dev-backend.s3.cern.ch/media/files/10913/10.1007/JHEP06(2015)171.xml"
        target="_blank"
      >
        <span
          aria-label="file-text"
          class="anticon anticon-file-text"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="file-text"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM504 618H320c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM312 490v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H320c-4.4 0-8 3.6-8 8z"
            />
          </svg>
        </span>
         XML
      </a>
      <a
        class="file-big mr-2"
        download=""
        href="https://scoap3-dev-backend.s3.cern.ch/media/files/10913/10.1007/JHEP06(2015)171_a.pdf"
        target="_blank"
      >
        <span
          aria-label="file-pdf"
          class="anticon anticon-file-pdf"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="file-pdf"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M531.3 574.4l.3-1.4c5.8-23.9 13.1-53.7 7.4-80.7-3.8-21.3-19.5-29.6-32.9-30.2-15.8-.7-29.9 8.3-33.4 21.4-6.6 24-.7 56.8 10.1 98.6-13.6 32.4-35.3 79.5-51.2 107.5-29.6 15.3-69.3 38.9-75.2 68.7-1.2 5.5.2 12.5 3.5 18.8 3.7 7 9.6 12.4 16.5 15 3 1.1 6.6 2 10.8 2 17.6 0 46.1-14.2 84.1-79.4 5.8-1.9 11.8-3.9 17.6-5.9 27.2-9.2 55.4-18.8 80.9-23.1 28.2 15.1 60.3 24.8 82.1 24.8 21.6 0 30.1-12.8 33.3-20.5 5.6-13.5 2.9-30.5-6.2-39.6-13.2-13-45.3-16.4-95.3-10.2-24.6-15-40.7-35.4-52.4-65.8zM421.6 726.3c-13.9 20.2-24.4 30.3-30.1 34.7 6.7-12.3 19.8-25.3 30.1-34.7zm87.6-235.5c5.2 8.9 4.5 35.8.5 49.4-4.9-19.9-5.6-48.1-2.7-51.4.8.1 1.5.7 2.2 2zm-1.6 120.5c10.7 18.5 24.2 34.4 39.1 46.2-21.6 4.9-41.3 13-58.9 20.2-4.2 1.7-8.3 3.4-12.3 5 13.3-24.1 24.4-51.4 32.1-71.4zm155.6 65.5c.1.2.2.5-.4.9h-.2l-.2.3c-.8.5-9 5.3-44.3-8.6 40.6-1.9 45 7.3 45.1 7.4zm191.4-388.2L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494z"
            />
          </svg>
        </span>
         PDF/A
      </a>
    </dd>
  </dl>
</div>
`;
